electrum/plugin.py

```diff
@@ -559,7 +570,7 @@ def force_pair_xpub(self, plugin: 'HW_PluginBase', handler: 'HardwareHandlerBase
         # choose an unpaired device and compare its first address.
         xtype = bip32.xpub_type(xpub)
         client = self._client_by_id(info.device.id_)
-        if client and client.is_pairable():
+        if client and client.is_pairable() and self._type_check_client_plugin(client, plugin):
```

Comment:
I think we should just do a direct comparison here
`type(client.plugin) == type(plugin)`
then there is no need for a helper function

Or do you think it might be useful to log this? I am not so sure...

Reply:
I still think that it has to be done in `client_by_xpub` too... then we have it on 2 places and helper functions fit there well

Reply:
Yes, I agree it is probably needed in `client_by_xpub` too.
In this case I am suggesting duplicating the code.
I am not against helper functions in general but don't see the need here.
For one, the direct comparison is fewer characters, so that in itself is a hint there is no point for the helper :P
`self._type_check_client_plugin(client, plugin)`
`type(client.plugin) == type(plugin)`